javascript/ql/lib/semmle/javascript/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -353,3 +353,23 @@ module ModelOutputSpecific {
     )
   }
 }
+
+/**
+ * Holds if the edge `pred -> succ` labelled with `path` exists in the API graph.
+ */
+bindingset[pred]
+predicate apiGraphHasEdge(API::Node pred, string path, API::Node succ) {
```

Comment:
No guarantees and no currently no tests. I spent a bit of time refactoring the `ModelOutput::getAWarning` into a parameterised module so we could check these as well, but it gets a bit fiddly.

Some tests are going to exist in an internal repo where we use model generation, in the sense that those tests would fail if there was a mismatch here.

Reply:
I think it would be nice with these kinds of tests, however I'm going to block this PR to get them in NOW.

Reply:
> however I'm going to block this PR to get them in NOW.

I assume a "not" was accidentally left out of that sentence 😅

Reply:
indeed :see_no_evil: (fixed by edit)